apps/opik-backend/src/test/java/com/comet/opik/api/resources/v1/priv/PromptResourceTest.java

```diff
@@ -1091,6 +1427,167 @@ Stream<Arguments> when__promptVersionIsInvalid__thenReturnError() {
         }
     }
 
+    @Nested
+    @DisplayName("Get Prompt Versions by Prompt Id")
+    @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+    class GetPromptVersionsByPromptId {
+
+        @Test
+        @DisplayName("Success: should get prompt versions by prompt id")
+        void shouldGetPromptVersionsByPromptId() {
+
+            var prompt = factory.manufacturePojo(Prompt.class).toBuilder()
+                    .lastUpdatedBy(USER)
+                    .createdBy(USER)
+                    .template(null)
+                    .build();
+
+            UUID promptId = createPrompt(prompt, API_KEY, TEST_WORKSPACE);
```

Comment:
IMO worth to create an additional prompt with version (e.g. `otherPrompt`) and this way your existing assertion verifies that only the versions that belong to `prompt` are returned